### ISSUE TITLE
Calling a function with the wrong number of arguments isn't an error

### ISSUE THREAD

**Issue by g-rocket** [processing-js/processing-js#195]
Example:

    void setup() {
      foo();
      foo(1,5);
    }
    
    void draw() {}
    
    void foo(int arg) {
      println(arg);
    }

I would expect both invocations of `foo` above to cause an error, and print something to the error console. Instead, they both succeed, the first one acting like `foo(undefined)` and the second one acting like `foo(1)`. I would argue this isn't desirable behavior, because it can turn simple typos into difficult-to-track-down bugs.

**Comment by GoToLoop** [comment]
* Your example is what can be called cross-mode code.
* 1st we run it under "Java Mode" in order to catch any obvious errors.
* Only then switch to "JavaScript Mode" as to assure it can be successfully transpired into JS too.
* It's common knowledge that JavaScript doesn't throw exceptions for the number of arguments passed.

**Comment by Pomax** [comment]
This is a consequence of working in JavaScript, which has no separation of variables and function names, and doesn't support function overloading - it simply has one function handler that takes however many arguments you throw at it. This difference, and other differences, are covered in the http://processingjs.org/articles/p5QuickStart.html article, as well as in the chapter on Processing.js in volume one of "the Architecture of Open Source Software), free to read at http://www.aosabook.org/en/pjs.html

TL;DR for this is that JavaScript is really very different from Java, and so the philosophy behind Pjs as a compiler is to assume you're giving it syntactically valid Processing code, without all the code for type checking, "identical errors to Java" error reporting, separated heaps for variables and functions, etc. All those things would slow down execution too much, given how Pjs started life and evolved over the last five years.

(A modern revisit using a C++ based transpiler, turned into a JS library using empscripten, for instance, might be fast enough, but that'd basically be a completely new project)
